magento/magento2
# Send order email for Braintree Paypal orders

### Description
This commits adds a frontend observer for sending new order emails.

Order emails are sent using observers on the
`sales_model_service_quote_submit_success` event, but only in the
`webapi_rest` and `webapi_soap` areas. When customers checkout with
Braintree Paypal the order placement occurs in the `frontend` area with
a request to `Magento\Braintree\Controller\Paypal\PlaceOrder`, so these
orders were not sent a new order email.

### Fixed Issues (if relevant)
1. magento/magento2#13778 Braintree Paypal Method No Order Confirmation Email Sent
2. magento/magento2#12792 [2.1.10] No order confirmation email after paying with PayPal Express

### Manual testing scenarios
1. Enable Braintree including PayPal through Braintree
2. Complete PayPal purchase on store frontend. The redirect back to the store from PayPal should be to `braintree/paypal/review`
3. Completing the order place process should result in final request to `braintree/paypal/placeorder`
4. Will receive the new order email

### Contribution checklist
